File: crates/ferrule-codemap/tests/fixtures/ts/src/price.ts
export class Money {
  constructor(public cents: number) {}
}

export type Percent = number;

export enum Currency {
  Shekel,
  Euro,
}

export const MAX_PERCENT = 100;

export function applyDiscount(total: Money, percent: Percent): Money {
  return new Money((total.cents * (MAX_PERCENT - percent)) / MAX_PERCENT);
}

File: crates/ferrule-codemap/tests/fixtures/rust/src/price.rs
#[derive(Clone, Copy)]
pub struct Money {
    pub cents: i64,
}

impl Money {
    pub fn new(cents: i64) -> Self {
        Money { cents }
    }
}

pub fn apply_discount(total: Money, percent: i64) -> Money {
    Money::new(total.cents * (100 - percent) / 100)
}

pub enum Currency {
    Shekel,
    Euro,
}

macro_rules! money {
    ($c:expr) => {
        Money::new($c)
    };
}
